lib/serialport.js

```diff
@@ -55,6 +52,29 @@ var defaultSetFlags = {
   rts: true
 };
 
+function assign(target, defaults, user) {
```

Comment:
My focus wasn't to strictly emulate `object.assign()`, it was to serve the needs of this particular module.